cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -182,9 +198,9 @@ public void shutdown() {
     KafkaCruiseControlUtils.closeAdminClientWithTimeout(_adminClient);
 
     try {
-      _detectorScheduler.awaitTermination(_anomalyDetectionIntervalMs, TimeUnit.MILLISECONDS);
+      _detectorScheduler.awaitTermination(_goalViolationDetectionIntervalMs, TimeUnit.MILLISECONDS);
```

Comment:
It is not clear why the `_detectorScheduler` termination time is associated with `_goalViolationDetectionIntervalMs`.

Reply:
Set to the max of all intervals.